pkg/util/ecs/metadata/testutil/dummy_ecs.go

```diff
@@ -72,6 +100,9 @@ func NewDummyECS(ops ...Option) (*DummyECS, error) {
 // ServeHTTP is used to handle HTTP requests.
 func (d *DummyECS) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	fmt.Printf("dummyECS received %s on %s\n", r.Method, r.URL.Path)
+	d.Lock()
+	d.RequestCount++
+	d.Unlock()
```

Comment:
As suggested above, the use of `atomic` values can avoid the need to lock the mutex here.

```suggestion
	d.RequestCount.Add(1)
```